design/baremetal-operator/reboot-interface.md

```diff
@@ -138,6 +138,26 @@ The controller automatically removes all annotations with the
 
 - the Host is deprovisioned
 
+The API has been extended to permit either a hard or a soft reboot to
+take place, specified via the reboot annotation and a `mode` parameter. The API
+supports the following syntax:
+
+- `reboot.metal.io` -- immediate reboot via soft shutdown first,
+  followed by a hard shutdown if the soft shutdown fails.
+- `reboot.metal3.io: {'mode':'hard'}` -- immediate reboot via hard
+  shutdown, potentially allowing for high-availability use-cases.
+- `reboot.metal3.io/{key}` -- phased reboot, issued and managed by the
+  client registered with the `key`, via soft shutdown first, followed
+  by a hard reboot if the soft reboot fails.
+- `reboot.metal3.io:/{key} {'mode':'hard'}` -- phased reboot, issued
+  and managed by the client registered with the `key`, via hard
+  shutdown.
+
+Currently, if any mode is specified other than `{'mode':'hard'}`, the default behaviour
```

Comment:
The way I read it is that `{'mode':'foo'}` will trigger soft reboot, but there's no description for any other value, such as just `foo`.
There are already clients who are using this annotation with some value (e.g. OCP UI), so maybe I'd replace `mode` with `value`.

Reply:
The annotation, and values/schema supported should be owned by the BMH/BMO - it doesn't seem right that external systems are (ab)using the annotation to store state - how does changing the key from `mode` to `value` help here if we have no control over whatever arbitrary data may be written?

Reply:
Thanks, have clarified.